cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -323,14 +318,57 @@ open class Node @Inject constructor(private val project: Project) : CordformNode
                 .withValue("rpcSettings.address", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.port}"))
                 .withValue("rpcSettings.adminAddress", ConfigValueFactory.fromAnyRef("$containerName:${rpcSettings.adminPort}"))
                 .withValue("detectPublicIp", ConfigValueFactory.fromAnyRef(false))
-        val tmpConfFile = createTempConfigFile(dockerConf.root())
+
+        createNodeAndWebServerConfigFiles(dockerConf)
+    }
+
+    private fun createNodeAndWebServerConfigFiles(config: Config) {
+        val tmpConfFile = createTempConfigFile(config.toNodeOnly().root(), "node.conf")
         appendOptionalConfig(tmpConfFile)
         project.copy {
             it.apply {
                 from(tmpConfFile)
                 into(rootDir)
             }
         }
+        if (config.hasPath("webAddress")) {
+            val webServerConfigFile = createTempConfigFile(config.toWebServerOnly().root(), "web-server.conf")
+            project.copy {
+                it.apply {
+                    from(webServerConfigFile)
+                    into(rootDir)
+                }
+            }
+        }
+    }
+
+    private fun Config.toNodeOnly(): Config {
+        return if (hasPath("webAddress")) {
+            withoutPath("webAddress").withoutPath("useHTTPS")
+        } else {
+            this
+        }
+    }
+
+    private fun Config.toWebServerOnly(): Config {
+        var config = ConfigFactory.empty()
```

Comment:
Can you rename this `newConfig` or something? It's a bit confusing to read otherwise. Might also be worth chaining these operations in some way.